### PR TITLE
Don't rely completely on wp_enqueue_editor or wp_enqueue_media

### PR DESCRIPTION
**Pull request by roborourke**
This is a first pass at moving to something less coupled to the editor or the media libraries. Once shortcake is fully decoupled from the media popup the `wp_enqueue_media()` can be removed from `Shortcode_UI\enqueue()`.

Fieldmanager is a bit annoying as the field classes available to shortcake have to be initialised whether shortcake is rendered or not but at least the templates aren't output. This is because Fieldmanager assumes all the field classes are created before the `admin_enqueue_scripts` action - `wp_enqueue_editor` is bound to `admin_print_footer_scripts` so there's no other way around it yet.

Addresses #153

### CODE REVIEW

File: shortcode-ui.php [hunk 1]
```diff
@@ -58,3 +58,10 @@ function shortcode_ui_register_for_shortcode( $shortcode_tag, $args = array() )
 function shortcode_ui_get_register_shortcode( $shortcode_tag, $args = array() ) {
 	return Shortcode_UI::get_instance()->get_shortcode( $shortcode_tag );
 }
+
+/**
+ * Queue the shortcode UI scripts & templates manually
+ */
+function enqueue_shortcode_ui() {
```
Comment (danielbachhuber) [inline shortcode-ui.php]: Let's call this `shortcode_ui_enqueue_assets()` to properly namespace

Comment (roborourke) [inline shortcode-ui.php]: Sorted

**Comment by roborourke** [comment]
@mattheu @danielbachhuber need me to do anything else on this or do you have another approach in mind?

**Comment by danielbachhuber** [comment]
Well, I'd like to see them decoupled (e.g. media library code independent of TinyMCE views and vice versa). Not sure what @mattheu is currently thinking though.

**Comment by roborourke** [comment]
@danielbachhuber at the moment this couples the output by default to the `wp_enqueue_editor` action as the plugin did originally but without relying on `print_media_templates`. It fixes a bunch of compatibility issues I've run into with things like the Black Studio MCE widget.

When someone needs to use the plugin in another context the `shortcode_ui_enqueue_assets()` function can be called so once the UI is decoupled from the media modal it should just continue to work using this function. We'd only need to remove the call to `wp_enqueue_media()` in the enqueue method.

**Comment by mattheu** [comment]
Someone else has actually run into this exact same issue.

I think we should go ahead and merge this regardless as it fixes an issue. I'll take a look into decoupling the JS a bit further as a separate task.

File: inc/class-shortcode-ui.php [hunk 2]
```diff
@@ -137,10 +155,12 @@ public function action_wp_enqueue_editor() {
 	 *
 	 * @return null
 	 */
-	public function action_print_media_templates() {
+	public function print_templates() {
```
Comment (danielbachhuber) [inline inc/class-shortcode-ui.php]: Can we keep this name consistent?

**Comment by danielbachhuber** [comment]
Sounds good. One minor nit.

File: inc/class-shortcode-ui.php (hunk 2 above)
Comment (roborourke) [inline inc/class-shortcode-ui.php]: Cheers @danielbachhuber, makes sense. Should be good now.

Comment (roborourke) [inline inc/class-shortcode-ui.php]: D'oh - making this consistent with the action name it's hooked to rather than just reverting the name.

**Pull request merged**